web/src/engine/predictive-text/worker-thread/src/main/correction/context-transition.ts

```diff
@@ -0,0 +1,105 @@
+import { ContextState } from './context-state.js';
+
+import { LexicalModelTypes } from '@keymanapp/common-types';
+import Distribution = LexicalModelTypes.Distribution;
+import Transform = LexicalModelTypes.Transform;
+
+/**
+ * Represents the transition between two context states as triggered
+ * by input keystrokes or applied suggestions.
+ */
+export class ContextTransition {
+  /**
+   * Represents the state of the context before the transition event occurred.
+   */
+  readonly base: ContextState;
+  private _final: ContextState;
+
+  /**
+   * Indicates the fat-finger distribution for the incoming keystroke related to
+   * the context transition event.
+   */
+  inputDistribution?: Distribution<Transform>;
+
+  // The transform ID in play.
+  private _transitionId?: number;
```

Comment:
Why transition vs transform here? The comment says 'transform', the variable says 'transition'...

Also, the comment doesn't really say much to explain what this variable is for or what it does

Reply:
Sounds good, I've added #14515 based on your comment above.